src/c++/tests/client_timeout_test.cc

```diff
@@ -45,8 +45,114 @@ namespace tc = triton::client;
+  cudaIpcMemHandle_t cuda_shm_handle;
+  size_t count = 0;
+
+  COUNT_ERROR_MSGS(
+      grpc_client->IsServerLive(&success, headers, timeout_ms),
+      "Failed on IsServerLive", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->IsServerReady(&success, headers, timeout_ms),
+      "Failed on IsServerReady", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->IsModelReady(&success, model_name, "", headers, timeout_ms),
+      "Failed on IsModelReady", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->ServerMetadata(&server_metadata, headers, timeout_ms),
+      "Failed on ServerMetadata", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->ModelMetadata(
+          &model_metadata, model_name, "", headers, timeout_ms),
+      "Failed on ModelMetadata", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->ModelConfig(
+          &model_config, model_name, "", headers, timeout_ms),
+      "Failed on ModelConfig", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->ModelRepositoryIndex(&repository_index, headers, timeout_ms),
+      "Failed on ModelRepositoryIndex", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->ModelInferenceStatistics(
+          &infer_stat, model_name, "", headers, timeout_ms),
+      "Failed on ModelInferenceStatistics", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->LoadModel(model_name, headers, "", {}, timeout_ms),
+      "Failed on LoadModel", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->UnloadModel(model_name, headers, timeout_ms),
+      "Failed on UnloadModel", count);
+
+  COUNT_ERROR_MSGS(
+      grpc_client->UpdateTraceSettings(
+          &response, model_name, settings, headers, timeout_ms),
+      "Failed on UpdateTraceSettings", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->GetTraceSettings(
+          &trace_settings, model_name, headers, timeout_ms),
+      "Failed on GetTraceSettings", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->SystemSharedMemoryStatus(
+          &shmstatus, memory_name, headers, timeout_ms),
+      "Failed on SystemSharedMemoryStatus", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->RegisterSystemSharedMemory(
+          memory_name, memory_name, byte_size, 0, headers, timeout_ms),
+      "Failed on RegisterSystemSharedMemory", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->UnregisterSystemSharedMemory(
+          memory_name, headers, timeout_ms),
+      "Failed on UnregisterSystemSharedMemory", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->CudaSharedMemoryStatus(
+          &cuda_shmstatus, "", headers, timeout_ms),
+      "Failed on CudaSharedMemoryStatus", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->RegisterCudaSharedMemory(
+          model_name, cuda_shm_handle, 0, byte_size, headers, timeout_ms),
+      "Failed on RegisterCudaSharedMemory", count);
+  COUNT_ERROR_MSGS(
+      grpc_client->UnregisterCudaSharedMemory(memory_name, headers, timeout_ms),
+      "Failed on UnregisterSystemSharedMemory", count);
+  if (count > 0 && count != 18) {
```

Comment:
Why do we need two conditional checks on the same variable?
Shouldn't `count != 18` be enough?

Reply:
There's also a long timeout test, therefore the count should be == 0. I can reflect that so it's less confusing

Reply:
fixed the whole test so this is not relevant anymore...